### PR TITLE
feat: refactor interoperability section

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup can you give a quick round of feedback? I suggest not to read the diff, but just the new text.

File: src/creating-objects.md [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
-newFile("HelloWorld.txt")
+import java.io.File
+
+def main(): Unit \ IO = 
+    let f = new File("foo.txt");
+    println("Hello World!")
+```
+
+Here we import the `java.io.File` class and instantiate a `File` object by
+calling one of its constructor using the `new` keyword.
```
Comment (mlutze) [inline src/creating-objects.md:18]: constructor -> constructors

**Review by mlutze — COMMENTED** [review]
I suspect a common error is where type inference fails. Should we have a section for that?

File: src/calling-methods.md [hunk 2]
```diff
@@ -1,64 +1,128 @@
...
 ```
 
+Here we know for certain that `Math.pow` and `Math.sqrt` are _pure_ functions,
+hence we can put them inside an `unsafe` block. Thus we are able to type check
+the Flix `pythagoras` function as pure, i.e. without the `IO` effect.
+
+> **Warning:** Do not, under any circumstances, use `unsafe` on expressions that
+> have side-effects. Doing so breaks the type and effect system which can lead
+> to incorrect compiler optimizations which can change the meaning of your
+> program in subtle or catastrophic ways!
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply src/calling-methods.md:128]: Will do, 2x

**Pull request merged**